Google.PowerShell.IntegrationTests/SQL/SQL.GcSqlInstances.Tests.ps1

```diff
@@ -73,6 +73,24 @@ Describe "Add-GcSqlInstance" {
         $myInstance.Settings.MaintenanceWindow.Hour | Should Be 2
         gcloud sql instances delete $instance --quiet 2>$null
     }
+
+    It "should be able to create a read replica instance" {
+        $r = Get-Random
+        # A random number is used to avoid collisions with the speed of creating
+        # and deleting instances.
+        $instance = "test-inst$r"
+        $setting = New-GcSqlSettingConfig "db-n1-standard-1" 
+        $replicaConfig = New-GcSqlInstanceReplicaConfig
+        $config = New-GcSqlInstanceConfig $instance -SettingConfig $setting `
+            -ReplicaConfig $replicaConfig -MasterInstanceName "test-db2"
+        Add-GcSqlInstance $config
+        $newInstances = Get-GcSqlInstance
+        ($newInstances.Name -contains $instance) | Should Be true
+        $myInstance = Get-GcSqlInstance $instance
+        $myInstance.MasterInstanceName | Should Be "gcloud-powershell-testing:test-db2"
+        $myInstance.InstanceType | Should Be "READ_REPLICA_INSTANCE"
+        gcloud sql instances delete $instance --quiet 2>$null
```

Comment:
It is possible that you could have a cycle where cmdlet A calls cmdlet B calls cmdlet A, and if there is a bug in either of them then the test could be considered invalid.

However, I assume that `Remove-GcSqlInstance` has a separate set of tests that work great. And so it might make sense to just use that rather than rely on `gcloud`.